TAF_EasyRest/TestFramework/Pages/Client/OrderHistoryPage.cs

```diff
@@ -9,52 +10,65 @@ public class OrderHistoryPage : BasePage
         public PersonalInfoPageComponent PersonalInfoPageComponent { get; }
         public List<OrderPageComponent> orders { get; set; }
+        private string _pageUrl => GetUrls.getUrl("OrderHistoryPage").Url;
 
-        public OrderHistoryPage(IWebDriver driver): base(driver)
+        public OrderHistoryPage(IWebDriver driver) : base(driver)
         {
             NavigationMenuPageComponent = new NavigationMenuPageComponent(driver);
             PersonalInfoPageComponent = new PersonalInfoPageComponent(driver);
-            FillOdersList();
+
         }
 
         #region Elements
 
-        private IWebElement _allButton => driver.FindElement(By.XPath("//span[contains(text(),'All')]/ancestor::a"));
-        private IWebElement _historyButton => driver.FindElement(By.XPath("//span[contains(text(),'History (')]/ancestor::a"));
-                
+        private By _allButton => By.XPath("//span[contains(text(),'All')]/ancestor::a");
+        private By _historyButton => By.XPath("//span[contains(text(),'History (')]/ancestor::a");
+
         #endregion
 
         #region Methods
 
-        private int СountOrders()
+        public int CountOrders(int timeToWait)
         {
-            IReadOnlyCollection<IWebElement> items = driver.FindElements(By.XPath("//div[@class='MuiButtonBase-root-106 MuiExpansionPanelSummary-root-573']"));
+            driver.WaitUntilElementIsVisible(By.XPath("//div[contains(@class,'MuiExpansionPanel-rounded')]"), timeToWait);
+
+            IReadOnlyCollection<IWebElement> items = driver.FindElements(By.XPath("//div[contains(@class,'MuiExpansionPanel-rounded')]"));
             return items.Count();
         }
 
         private void FillOdersList()
         {
-            orders = new List<OrderPageComponent>(СountOrders());
-            for (int i = 0; i < orders.Count; i++)
+            int countOrders = CountOrders(3);
+
+            orders = new List<OrderPageComponent>();
+            for (int i = 0; i < countOrders; i++)
             {
-                orders.Add(new OrderPageComponent(driver, (i+1)));
+                orders.Add(new OrderPageComponent(driver, (i + 1)));
             }
         }
 
-        public OrderHistoryPage ClickAllButton()
+        public OrderHistoryPage ClickAllButton(int timeToWait)
         {
-            _allButton.Click();
+            driver.WaitUntilElementIsVisible(_allButton, timeToWait)
+                .Click();
             FillOdersList();
             return this;
         }
 
-        public OrderHistoryPage ClickHistoryButton()
+        public OrderHistoryPage ClickHistoryButton(int timeToWait)
         {
-            _historyButton.Click();
+            driver.WaitUntilElementIsVisible(_historyButton, timeToWait)
+                .Click();
             FillOdersList();
             return this;
         }
 
+        public override void GoToUrl()
+        {
+            Console.WriteLine(baseUrl + _pageUrl);
```

Comment:
why do you log into Console if you have a report for that purpose?

Reply:
Funny moment, my bad. This method has to do driver.Navigate.... Thx for comment.